### PR TITLE
Instance group wait for status

### PR DESCRIPTION
**Pull request by slevenick**
Fixes: https://github.com/hashicorp/terraform-provider-google/issues/8311

If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
compute: added support for output-only `status` field on `google_compute_instance_group_manager` and `google_compute_region_instance_group_manager`
```
```release-note:enhancement
compute: added support for `wait_for_instances_status` on `google_compute_instance_group_manager` and `google_compute_region_instance_group_manager`
```

### CODE REVIEW

File: mmv1/third_party/terraform/resources/resource_compute_instance_group_manager.go.erb [hunk 1]
```diff
@@ -273,6 +273,13 @@ func resourceComputeInstanceGroupManager() *schema.Resource {
 				Default:     false,
 				Description: `Whether to wait for all instances to be created/updated before returning. Note that if this is set to true and the operation does not succeed, Terraform will continue trying until it times out.`,
 			},
+			"wait_for_instances_status": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "STABLE",
```
Comment (rileykarson) [inline mmv1/third_party/terraform/resources/resource_compute_instance_group_manager.go.erb:279]: I think we need a check on this value in Read- if it's `""`, set it to `"STABLE"`. That way we get away without a state upgrader and don't have a one-off diff for resources provisioned with older provider versions.

File: mmv1/third_party/terraform/website/docs/r/compute_instance_group_manager.html.markdown [hunk 2]
```diff
@@ -129,6 +129,11 @@ The following arguments are supported:
...
     continue trying until it times out.
 
+* `wait_for_instances_status` - (Optional) When used with `wait_for_instances` it specifies the status to wait for.
+    When `STABLE` is specified this resource will wait until the instances are stable before returning. When `UPDATED` is
+    set, it will wait for the version target to be reached and any per instance configs to be effective as well as all
+    instances to be stable before returning. The possible values are `STABLE` and `UPDATED`
+
 ---
 
 * `auto_healing_policies` - (Optional) The autohealing policies for this managed instance
```
Comment (rileykarson) [inline mmv1/third_party/terraform/website/docs/r/compute_instance_group_manager.html.markdown:139]: We want to document the attribute fields under `status` too, right?

File: mmv1/third_party/terraform/tests/resource_compute_disk_test.go.erb [hunk 3]
```diff
@@ -790,6 +790,7 @@ resource "google_compute_instance_group_manager" "manager" {
   // block on instances being ready so that when they get deleted, we don't try
   // to continue interacting with them in other resources
   wait_for_instances = true
+  wait_for_instances_status = "UPDATED"
```
Comment (rileykarson) [inline mmv1/third_party/terraform/tests/resource_compute_disk_test.go.erb:793]: Is this changed due to a failing test, or just to ensure we use it somewhere? If it's just to use it, can we do so in an IGM-specific test?

Comment (rileykarson) [reply mmv1/third_party/terraform/tests/resource_compute_disk_test.go.erb:793]: Also, can that test migrate from one to the other?

**Comment by slevenick** [comment]
> LGTM- if you don't mind provisioning a MIG with an old version and then updating to the dev provider including this change locally, that will ensure that we handle the state transition correctly.

Yup, wait_for_instances_status gets set to STABLE in state with no diff

**Pull request merged**